x/ibc-transfer/keeper/relay.go

```diff
@@ -70,17 +71,36 @@ func (k Keeper) SendTransfer(
 
 	// begin createOutgoingPacket logic
 	// See spec for this logic: https://github.com/cosmos/ics/tree/master/spec/ics-020-fungible-token-transfer#packet-relay
-
 	channelCap, ok := k.scopedKeeper.GetCapability(ctx, host.ChannelCapabilityPath(sourcePort, sourceChannel))
 	if !ok {
 		return sdkerrors.Wrap(channeltypes.ErrChannelCapabilityNotFound, "module does not own channel capability")
 	}
 
+	prefixedDenom := token.Denom
+
+	// NOTE: denomination and hex hash correctness checked during msg.ValidateBasic
+	if strings.HasPrefix(token.Denom, "ibc/") {
+		hexHash := token.Denom[4:]
+		hash, err := types.ParseHexHash(hexHash)
+		if err != nil {
+			return err
+		}
+
+		denomTrace, found := k.GetDenomTrace(ctx, hash)
+		if !found {
+			return sdkerrors.Wrap(types.ErrTraceNotFound, hexHash)
+		}
+		prefixedDenom = denomTrace.GetPrefix() + denomTrace.BaseDenom
+	} else if strings.Contains(token.Denom, "/") {
```

Comment:
addition to spec to allow users to pass in either the `ibc/{hash}` denom or the trace prefixed denomination

Reply:
hmm, honestly I would prefer that we require one or the other instead of putting conditional logic in the state machine

Reply:
Agreed, this flexibility should be at the UI level not state-machine level